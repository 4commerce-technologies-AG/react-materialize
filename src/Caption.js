import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import constants from './constants';

const Caption = ({ className, placement, children, ...props }) => {
  return (
    <div
      className={cx(
        'caption',
        {
          [`${placement}-align`]: placement
        },
        className
      )}
      {...props}
    >
      {children}
    </div>
  );
};

Caption.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * Placement of the caption
   * @default 'center'
   */
  placement: PropTypes.oneOf(constants.PLACEMENTS),
  style: PropTypes.object
};

Caption.propDefault = {
  placement: 'center'
};

export default Caption;
