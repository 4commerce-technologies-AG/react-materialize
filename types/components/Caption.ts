import * as React from 'react';

declare namespace Caption {
}

declare class Caption extends React.Component {}

export = Caption;
