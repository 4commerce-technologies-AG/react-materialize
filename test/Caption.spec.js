import React from 'react';
import { shallow } from 'enzyme';
import Caption from '../src/Caption';

describe('<Caption />', () => {
  let wrapper = shallow(
    <Caption placement="left">
      <h3>This is our big Tagline!</h3>
      <h5 className="light grey-text text-lighten-3">
        Here's our small slogan.
      </h5>
    </Caption>
  );

  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
