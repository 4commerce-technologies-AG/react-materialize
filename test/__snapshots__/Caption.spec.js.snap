// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption /> renders 1`] = `
<div
  className="caption left-align"
>
  <h3>
    This is our big Tagline!
  </h3>
  <h5
    className="light grey-text text-lighten-3"
  >
    Here's our small slogan.
  </h5>
</div>
`;
